Support Dropbox v2 api

suwatch opened the pull request:
This is to support synchng using OAuth 2.0 and Dropbox v2 APIs.

suwatch commented on Kudu.Services/ServiceHookHandlers/DropboxHandler.cs:
```diff
@@ -37,7 +38,7 @@ public DeployAction TryParseDeploymentInfo(HttpRequestBase request, JObject payl
                 message = String.Format(CultureInfo.CurrentUICulture, Resources.Dropbox_SynchronizingNChanges, dropboxInfo.DeployInfo.Deltas.Count);
             }
-            else if (String.Equals(payload.Value<string>("scmType"), "DropboxV2", StringComparison.OrdinalIgnoreCase))
+            else if (String.Equals(payload.Value<string>(DropboxVersionKey), "2", StringComparison.OrdinalIgnoreCase))
```
Since we will not (don't need to) introduce a new ScmType, I explicitly using dropbox_version in the payload to signify the different version.

nickwalkmsft replied:
Is no one using an scmType of `DropboxV2`?

suwatch replied:
Correct.   BTW, this was done long time ago (but not completely correct) to anticipate the Dropbox V2.   Changing ScmType is unnecessary and may involve more codes than we need.

nickwalkmsft commented:
Can you summarize a bit?

This looks mostly like modifications as opposed to additions; is any functionality being lost, or are we currently compatible with the v2 API but just not using it to the fullest? If it's the latter, what does this add?

suwatch commented:
A couple years back when Dropbox V2 was available, we added V2 support in Kudu in anticipation of Antares change.   However, the way we implemented was not entirely correct.   

Dropbox V2 introduced the new OAuth 2.0 support as well as a set of new V2 REST apis.   Our implementation only adopted the OAuth 2.0 but still using the V1 REST apis.   Note that Dropbox V1 api supports both OAuth 1.0 and 2.0 tokens while Dropbox V2 only supports OAuth 2.0.

My change is to embrace both OAuth 2.0 and V2 REST api - so it looks more like a modification than a full addition.

davidebbo commented:
@suwatch isn't this already in?
